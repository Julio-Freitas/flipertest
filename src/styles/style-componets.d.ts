// import original module declarations
import 'styled-components';
import { theme } from './src/styles/theme';

type Theme = typeof theme;
declare module 'styled-components' {
    export interface DefaultTheme extends Theme {
        backgournd: string;
        colors: {
            primary: string;
            secondary: string;
            borderColor: string;
        };
        spacings: {
            small: string;
            medium: string;
            large: string;
        };
        font: {
            sizes: {
                small: string;
                medium: string;
                large: string;
            };
        };
    }
}
