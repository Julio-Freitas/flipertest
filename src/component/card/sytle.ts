import styled from 'styled-components';

type SizeTypes = 'small' | 'medium' | 'large' | undefined;
type FontWeight = 'normal' | 'bold';

type propsText = {
    primary?: boolean;
    fontSize?: SizeTypes;
    fontWeight?: FontWeight;
    paddingTop?: SizeTypes;
    paddingBottom?: SizeTypes;
    paddingRight?: SizeTypes;
    paddingLeft?: SizeTypes;
};

export const Header = styled.div`
    padding-top: 1em;
    display: flex;
    align-items: center;
    justify-content: space-between;
`;
export const Icon = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    position: relative;
    svg {
        cursor: pointer;
    }
`;
export const MenuIcon = styled.div`
    position: absolute;
    right: 14px;
    width: 100px;
    border: 1px solid #ccc;
    height: 22px;
    padding: 0 10px;
    font-weight: bold;
    display: flex;
    align-items: center;
    color: ${({ theme }) => theme.colors.primary};
    border-radius: 5px;
    background-color: ${({ theme }) => theme.backgournd};
    cursor: pointer;
    transition: ease-in 0.5s;
    &:hover {
        font-size: ${({ theme }) => theme.font.sizes.medium};
    }
`;

export const Content = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

export const Detatil = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    padding-top: 1.2em;
`;

export const Text = styled.span<propsText>`
    display: block;
    color: ${({ primary, theme }) => (primary ? theme.colors.primary : theme.colors.secondary)};
    font-size: ${({ fontSize, theme }) => !!fontSize && theme.font.sizes[fontSize]};
    font-weight: ${({ fontWeight }) => fontWeight || 'normal'};

    ${({ paddingTop, paddingRight, paddingBottom, paddingLeft, theme }) =>
        paddingTop
            ? `padding-top: ${theme.spacings[paddingTop]};`
            : paddingRight
            ? `padding-right: ${theme.spacings[paddingRight]};`
            : paddingBottom
            ? `padding-bottom: ${theme.spacings[paddingBottom]};`
            : !!paddingLeft && `padding-left: ${theme.spacings[paddingLeft]};`}
`;

export const Separator = styled.hr`
    border: 0.05em solid #9aa3bc5c;
    margin: 26px 0;
`;
