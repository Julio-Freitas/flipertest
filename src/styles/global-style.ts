import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
@import url('https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,300;0,400;1,300;1,400&display=swap');
*{
  margin:0;
  padding:0;
  outline:none;
}
  html{
    font-size: 62.5%
  }
  body{
    height: 100%;
    font-family: 'Open Sans', 'Helvetica Neue', sans-serif;
    background-color: ${({ theme }) => theme.backgournd};
  }
  a{
    text-decoration: opacity 300ms ease-in-out;
    text-decoration: none;

    &:hover{
      opacity: .6;
    }
  }
  button{
    cursor: pointer;
  }
`;
