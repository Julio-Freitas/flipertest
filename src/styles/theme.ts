export const theme = {
    backgournd: '#E8EBF6',
    colors: {
        primary: '#3B5CB8',
        secondary: '#606377',
        borderColor: '#9bb3cd',
    },

    spacings: {
        small: '1em',
        medium: '2em',
        large: '3em',
    },
    font: {
        sizes: {
            small: '1em',
            medium: '1.38em',
            large: '2em',
        },
    },
};
