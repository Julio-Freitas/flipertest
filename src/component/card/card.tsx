import { useState } from 'react';
import { ReactComponent as IconMore } from 'assests/svg/more_cion.svg';
import history from 'services/history';
import { Button } from '../button';

import * as Styled from './sytle';

export type PropsCommentsProps = {
    id: number;
    invested_value: string;
    amout?: number;
    rentability?: string;
    gain?: string;
    cdi?: string;
    datail?: boolean;
};
const Card = ({
    invested_value,
    rentability,
    gain,
    cdi,
    id,
    datail,
}: PropsCommentsProps): JSX.Element => {
    const [menu, setMenu] = useState(false);

    const redirectToId = (): void => {
        history.push('/detalhes', {
            id,
        });
        setMenu(false);
    };
    return (
        <div>
            <Styled.Header>
                <Styled.Text primary fontSize="large" fontWeight="bold">
                    Seu Resumo
                </Styled.Text>
                {!datail && (
                    <Styled.Icon
                        tabIndex={0}
                        onBlur={() => setMenu(false)}
                        status={menu}
                    >
                        <Styled.MenuIcon
                            onClick={redirectToId}
                            visibility={menu ? 'visible' : 'hidden'}
                        >
                            Ver card
                        </Styled.MenuIcon>

                        <IconMore
                            onClick={() => setMenu(!menu)}
                            data-testid="click-menu"
                        />
                    </Styled.Icon>
                )}
            </Styled.Header>
            <Styled.Content>
                <Styled.Text fontSize="large" paddingTop="small">
                    Valor investido
                </Styled.Text>
                <Styled.Text
                    fontSize="large"
                    primary
                    fontWeight="bold"
                    paddingTop="small"
                >
                    {invested_value}
                </Styled.Text>
                {rentability && (
                    <Styled.Detatil>
                        <Styled.Text fontSize="medium" paddingTop="small">
                            Rendabilidade/mês
                        </Styled.Text>
                        <Styled.Text
                            fontWeight="bold"
                            paddingTop="small"
                            fontSize="large"
                            primary
                        >
                            {rentability}
                        </Styled.Text>
                    </Styled.Detatil>
                )}

                {cdi && (
                    <Styled.Detatil>
                        <Styled.Text fontSize="medium" paddingTop="small">
                            CDI
                        </Styled.Text>
                        <Styled.Text fontWeight="bold" fontSize="large" primary>
                            {cdi}%
                        </Styled.Text>
                    </Styled.Detatil>
                )}

                {gain && (
                    <Styled.Detatil>
                        <Styled.Text fontSize="medium" paddingTop="small">
                            Ganho/mês
                        </Styled.Text>
                        <Styled.Text
                            fontWeight="bold"
                            paddingTop="small"
                            fontSize="large"
                            primary
                        >
                            {gain}
                        </Styled.Text>
                    </Styled.Detatil>
                )}
            </Styled.Content>
            <Styled.Separator />
            {datail && (
                <Button
                    type="button"
                    onClick={() => history.push('/dashboard')}
                    text="Dashboard"
                    data-after="Voltar à "
                    data-before=" Dashboard"
                />
            )}
        </div>
    );
};

export default Card;
