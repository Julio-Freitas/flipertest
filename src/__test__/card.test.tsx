import React from 'react';

import { theme } from '../styles/theme';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { Card } from '../component/card';
import history from '../services/history';

import { render, screen, fireEvent, cleanup } from '@testing-library/react';
beforeEach(() => cleanup());

const componentRender = (Component: React.Component, props): JSX.Element => {
    return (
        <ThemeProvider theme={theme}>
            <Component {...props} />
        </ThemeProvider>
    );
};

describe('Test component Card', () => {
    const Props = {
        id: 1,
        invested_value: 'R$3.200,00',
        amout: 5,
        rentability: 'R$10,00',
        gain: '5.0',
        cdi: '3.0',
        datail: false,
    };
    it('Test snapshot of Card', () => {
        const treeWithBreadCrumb = renderer.create(
            componentRender(Card, Props),
        );
        expect(treeWithBreadCrumb).toMatchSnapshot();
    });

    it('When click on icon show view the option (Ver Card)', async () => {
        render(componentRender(Card, Props));
        const option = await screen.getByTestId('click-menu');

        fireEvent.click(option);
        const textMeu = await screen.getByText(/ver card/i);
        expect(textMeu).toBeInTheDocument();
    });

    it('When click in (Ver Card) should redirect para path detalhes ', async () => {
        render(componentRender(Card, Props));
        const openMenu = await screen.getByTestId('click-menu');
        fireEvent.click(openMenu);
        const redirectTo = await screen.getByText(/ver card/i);
        fireEvent.click(redirectTo);
        expect(history.location.pathname).toBe('/detalhes');
    });
});
