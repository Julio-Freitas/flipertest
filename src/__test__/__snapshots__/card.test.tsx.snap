// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Card Test snapshot of Card 1`] = `
<div>
  <div
    className="sc-gtsrHT cwdsL"
  >
    <span
      className="sc-gKAaRy hhcfKr"
      fontSize="large"
      fontWeight="bold"
    >
      Seu Resumo
    </span>
    <div
      className="sc-dlnjwi sDJSC"
      onBlur={[Function]}
      tabIndex={0}
    >
      <svg
        data-testid="click-menu"
        onClick={[Function]}
      >
        more_cion.svg
      </svg>
    </div>
  </div>
  <div
    className="sc-eCApnc jnBVpW"
  >
    <span
      className="sc-gKAaRy gzygRA"
      fontSize="large"
    >
      Valor investido
    </span>
    <span
      className="sc-gKAaRy blqEoM"
      fontSize="large"
      fontWeight="bold"
    >
      R$3.200,00
    </span>
    <div
      className="sc-jSFjdj gHXOVH"
    >
      <span
        className="sc-gKAaRy iPyIma"
        fontSize="medium"
      >
        Rendabilidade/mês
      </span>
      <span
        className="sc-gKAaRy blqEoM"
        fontSize="large"
        fontWeight="bold"
      >
        R$10,00
      </span>
    </div>
    <div
      className="sc-jSFjdj gHXOVH"
    >
      <span
        className="sc-gKAaRy iPyIma"
        fontSize="medium"
      >
        CDI
      </span>
      <span
        className="sc-gKAaRy hhcfKr"
        fontSize="large"
        fontWeight="bold"
      >
        3.0
        %
      </span>
    </div>
    <div
      className="sc-jSFjdj gHXOVH"
    >
      <span
        className="sc-gKAaRy iPyIma"
        fontSize="medium"
      >
        Ganho/mês
      </span>
      <span
        className="sc-gKAaRy blqEoM"
        fontSize="large"
        fontWeight="bold"
      >
        5.0
      </span>
    </div>
  </div>
  <hr
    className="sc-iCoGMd hTrwoP"
  />
</div>
`;
